### PR TITLE
fix: flush insert data with nil buffer

### PR DESCRIPTION
**Pull request by aoiasd**
relate: https://github.com/milvus-io/milvus/issues/31165

### CODE REVIEW

File: internal/datanode/flow_graph_insert_buffer_node.go [hunk 1]
```diff
@@ -384,8 +384,9 @@ func (ibNode *insertBufferNode) FillInSyncTasks(fgMsg *flowGraphMsg, seg2Upload
 	// the aim for taking all these actions is to guarantee that the memory consumed by delBuf will not exceed a limit
 	segmentsToFlush := ibNode.delBufferManager.ShouldFlushSegments()
 	for _, segID := range segmentsToFlush {
+		buf := ibNode.GetBuffer(segID)
 		syncTasks[segID] = &syncTask{
-			buffer:    nil, // nil is valid
+			buffer:    buf, // nil is valid
 			segmentID: segID,
 		}
 	}
```
Comment (congqixia) [inline internal/datanode/flow_graph_insert_buffer_node.go:392]: maybe merge `segmentsToFlush` and `syncSegmentIDs` below is a better choice to keep code simple and safe

File: internal/datanode/flow_graph_insert_buffer_node.go [hunk 2]
```diff
@@ -382,25 +383,19 @@ func (ibNode *insertBufferNode) FillInSyncTasks(fgMsg *flowGraphMsg, seg2Upload
...
 	// the aim for taking all these actions is to guarantee that the memory consumed by delBuf will not exceed a limit
-	segmentsToFlush := ibNode.delBufferManager.ShouldFlushSegments()
-	for _, segID := range segmentsToFlush {
-		syncTasks[segID] = &syncTask{
-			buffer:    nil, // nil is valid
-			segmentID: segID,
-		}
-	}
+	segmentsToFlush := typeutil.NewUniqueSet(ibNode.delBufferManager.ShouldFlushSegments()...)
+	segmentsToFlush.Insert(ibNode.channel.listSegmentIDsToSync(fgMsg.endPositions[0].Timestamp)...)
```
Comment (bigsheeper) [inline internal/datanode/flow_graph_insert_buffer_node.go:387]: Synced segments do not necessarily require flushing. I thinkl we cannot set `task.flushed = true` for these segments.

Comment (congqixia) [reply internal/datanode/flow_graph_insert_buffer_node.go:387]: @bigsheeper `segmentsToFlush` here seem to be `segmentsToSync`, they are not set `Flushed=true`. Looks like naming issue?

Comment (bigsheeper) [reply internal/datanode/flow_graph_insert_buffer_node.go:387]: @congqixia Correct me if I'm wrong, I think they will be set at line 425:
```Golang
	mergeSyncTask(flushedSegments, syncTasks, func(task *syncTask) {
		task.flushed = true
	})
```

Comment (aoiasd) [reply internal/datanode/flow_graph_insert_buffer_node.go:387]: @bigsheeper flushedSegments was get at line 424
```
flushedSegments := ibNode.CollectSegmentsToSync()
```
means get manual sync segment from flushCh. Will not set segmentsToFlush

Comment (bigsheeper) [reply internal/datanode/flow_graph_insert_buffer_node.go:387]: Oh, sorry, my mistake.

Comment (congqixia) [reply internal/datanode/flow_graph_insert_buffer_node.go:387]: @bigsheeper actually it's naming issue, shall fix this later

**Review by congqixia — APPROVED** [review]
/approve

**Pull request merged**